src/test/java/io/strimzi/kafka/bridge/http/ConsumerSubscriptionTest.java

```diff
@@ -226,16 +241,23 @@ void listConsumerSubscriptions(VertxTestContext context) throws InterruptedExcep
 
         consumerService()
                 .createConsumer(context, groupId, json)
-                .subscribeConsumer(context, groupId, name, topic)
-                .subscribeConsumer(context, groupId, name, topic2);
-
-        CompletableFuture<Boolean> consume = new CompletableFuture<>();
+                .subscribeConsumer(context, groupId, name, topic, topic2);
 
         // poll to subscribe
+        CountDownLatch latch = new CountDownLatch(2);
         consumerService()
                 .consumeRecordsRequest(groupId, name, BridgeContentType.KAFKA_JSON_JSON)
-                .as(BodyCodec.jsonObject())
-                .send(ar -> consume.complete(true));
+                .as(BodyCodec.jsonArray())
+                .send(ar -> {
+                    latch.countDown();
+                });
+        consumerService()
+                .consumeRecordsRequest(groupId, name, BridgeContentType.KAFKA_JSON_JSON)
+                .as(BodyCodec.jsonArray())
+                .send(ar -> {
+                    latch.countDown();
+                });
+        latch.await(TEST_TIMEOUT, TimeUnit.SECONDS);
```

Comment:
can you try to avoid this double request for polling and the latch. I think we are just missing the `consume.get` and using the CompletableFuture in the right way.
See the `subscriptionConsumerDoesNotExistBecauseAnotherGroup` for example. There is the `subscribe` future that of course is waited with `subscribe.get` and completed in the lamba.